### ISSUE TITLE
Accessibility: Grouped form fields missing fieldset

### ISSUE THREAD

**Issue by vikrampm1** [woothemes/woocommerce#52085]
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

### Issue Description:

These form fields do not always have a visible label that provides context to the group of inputs by default - the component inserts a heading block above it, but this can be deleted by the content creator. Additionally, there is not a programmatically associated label for the controls in the group, so screen reader users do not get the context for what the filters are for.

![Image](https://github.com/user-attachments/assets/f4b00ff0-810f-4ef9-873b-c1d6028558d8)

### Component or Page: 

Component: Filter by attribute block

### Name affected element:

Filter by attribute block - list

### Relevant code snippets:

`<ul class="wc-block-checkbox-list wc-block-components-checkbox-list wc-block-attribute-filter-list"><li><div class="wc-block-components-checkbox wc-block-checkbox-list__checkbox"><label for="10"><input id="10" class="wc-block-components-checkbox__input" type="checkbox" aria-invalid="false" checked="">...</div></li></ul>`

### Recommended fix:

At a minimum, the checkboxes need to be wrapped in a <fieldset> tag and have a <legend> tag that describes the filters, such as <legend>Filter by $attribute_name</legend>. This can be screen reader only, but it would be best if it was visible on the page.  

A further enhancement on the block would be to include a heading tag in the legend and add a setting to the block that allows users to set the preferred heading level, H2-H6. Example: <legend><h2>Filter by $attribute_name</h2></legend> You want to allow users to be able to control the heading level as it may not always be correct for it to be set to an H2, however an H2 with be a good default. This can replace the inserted heading block whcih content creators can incorrectly remove. Or, if keeping the heading block either block the ability to remove it or warn users before they remove it. 

See: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset

**Comment by Manussakis** [comment]
I have started investigating/working on this issue and these are my thoughts:
 
- We should avoid changing the `<ul>` HTML tag. This can cause regressions on themes that expect it on their CSS selectors. Adding the `role="group"` attribute to it will make it behave like a `<fieldset>` element.
- The "Filter by attribute" title could be updated to "Filter by [attribute name]". e.g. "Filter by 'Color'".
- If changing the "Filter by attribute" title is not an option, the checkbox list could have another title only for screen readers.
- Making both titles visible may require design inputs:
  ![Image](https://github.com/user-attachments/assets/8fd2c61a-7519-4a84-acc7-aa4322651532)

I have created the https://github.com/woocommerce/woocommerce/pull/53681 draft PR with part of these updates and wanted to hear the opinion from the team who owns the Filter by Attribute block before moving forward.

cc: @vikrampm1

**Comment by Manussakis** [comment]
I came up with this approach, and I wanted to get feedback from the "Filter by attribute" block team before proceeding with the implementation.

### 1. Create a checkbox group
To make the checkbox list work as a group, the `<fieldset>` HTML tag should wrap it, but to avoid regressions by changing the markup, the solution is to add the `role="group"` to the [`<ul>` tag](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-blocks/packages/components/checkbox-list/index.tsx#L163).

### 2. Add a title to the checkbox group
It's not necessary to add a new title to the block markup, a title that already exists could be used to be connected to the `<ul>` element via the `aria-labelledby` attribute.

The "Filter by attribute" title seen when adding the block to the editor comes from the [`core/heading` block](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-blocks/assets/js/blocks/filter-wrapper/edit.tsx#L20) added to the Filter Wrapper block template.

![Image](https://github.com/user-attachments/assets/5815d27d-6c76-46da-9f50-4f67c0186cc0)

The problem is that it's hard to add a unique `id` attribute to the `core/heading` block and pass it along to the `woocommerce/attribute-filter` block from the Filter wrapper template.

The good news is that the `woocommerce/attribute-filter` block already has a heading with exactly the same appearance as the current heading.
![Image](https://github.com/user-attachments/assets/849b28f2-4ae5-44ef-b706-cdb844d3b701)

This means the `core/heading` block could be removed from the template and the heading from the `woocommerce/attribute-filter` could be used instead of using an [empty string](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-blocks/assets/js/blocks/filter-wrapper/edit.tsx#L26).

This would solve the problem of sharing the heading `id` value with the checkbox group as they are part of the same parent component:
- [Heading](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-blocks/assets/js/blocks/attribute-filter/block.tsx#L541)
- [Checkbox](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-blocks/assets/js/blocks/attribute-filter/block.tsx#L641)

I have created the https://github.com/woocommerce/woocommerce/pull/53681 draft PR with part of the implementation that helps to visualize the necessary changes.

If this approach is approved, the following blocks should be assessed as they seem to face similar accessibility issues.
- `woocommerce/active-filters`
- `woocommerce/price-filter`
- `woocommerce/stock-filter` https://github.com/woocommerce/woocommerce/issues/52130
- `woocommerce/attribute-filter`
- `woocommerce/rating-filter` https://github.com/woocommerce/woocommerce/issues/52128

cc: @vikrampm1

**Comment by vikrampm1** [comment]
@woocommerce/woo-fse can you help review the approach shared above and confirm before we proceed with the implementation?

**Comment by vikrampm1** [comment]
@woocommerce/woo-fse please let us know if you can provide a suggestion here or if we should proceed with the implementation and have you give feedback during the review?

**Comment by vikrampm1** [comment]
Closing this issue as per the discussion [here](https://github.com/woocommerce/woocommerce/pull/53681#issuecomment-2610425082).
